File: LoggerApp/app/src/main/java/experiments/com/pixellot/loggerapp/MainActivity.java
package experiments.com.pixellot.loggerapp;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import java.io.IOException;
import java.util.concurrent.TimeUnit;

import experiments.com.pixellot.logger.LogModel;
import experiments.com.pixellot.logger.Logger;

public class MainActivity extends AppCompatActivity {

    Logger logger;
    int counter = 0;
    long startTime;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        try {
            logger = new Logger(this);
            logger.setListener(new Logger.Listener() {
                @Override
                public void log(LogModel logModel) {
                    if(counter == 0)
                    {
                        startTime = System.nanoTime();
                        Log.d("LoggerResult","Start Logging:" + Thread.currentThread().getId());
                    }
                    counter++;
                    if (counter == 1000) {
                        Log.d("LoggerResult","End Logging.Time:" +TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
                        counter =0;
                    }
                }
            });
//            Logger.init(this);
        } catch (IOException e) {
            e.printStackTrace();
            Toast.makeText(this, "Please make sure you have permission to access file", Toast.LENGTH_LONG).show();
        }
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                testBenchmark();
            }
        });
    }

    private void testBenchmark() {
        long startTime = System.nanoTime();
        for (int i = 0; i < 1000; i++) {
            logger.d("Logger", "Logger->Message" + i);
        }
        Log.d("LoggerResult", "Logger ->Time:" + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));

//        startTime = System.nanoTime();
//        for (int i = 0; i < 1000; i++) {
//            Log.d("Logger", "Default->Message" + i);
//        }
//        Log.d("DefaultLoggerResult", "Time:" + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
    }
}

File: LoggerApp/app/src/main/java/experiments/com/pixellot/loggerapp/MainActivity.kt
package experiments.com.pixellot.loggerapp

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import android.view.View
import android.widget.Toast
import experiments.com.pixellot.logger.C
import experiments.com.pixellot.logger.LogModel
import experiments.com.pixellot.logger.Logger
import experiments.com.pixellot.logger.LoggerUtils.copyToExternalStorage
import java.io.IOException
import java.util.concurrent.TimeUnit

class MainActivity : AppCompatActivity() {
    internal lateinit var logger: Logger
    internal var counter = 0
    internal var startTime: Long = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        try {
            logger = Logger(this)
            logger.setListener(object : Logger.Listener {
                override fun log(logModel: LogModel) {
                    if (counter == 0) {
                        startTime = System.nanoTime()
                        Log.d("LoggerResult", "Start Logging:" + Thread.currentThread().id)
                    }
                    counter++
                    if (counter == SIZE) {
                        Log.d("LoggerResult", "End Logging.Time:" + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime))
                        counter = 0
                    }
                }
            })
            //            Logger.init(this);
        } catch (e: IOException) {
            e.printStackTrace()
            Toast.makeText(this, "Please make sure you have permission to access file", Toast.LENGTH_LONG).show()
        }

        findViewById<View>(R.id.button).setOnClickListener { testBenchmark() }
        findViewById<View>(R.id.copy).setOnClickListener { copy() }
    }

    private fun copy() {
        copyToExternalStorage(C.DB_NAME)
        Toast.makeText(this, "File is copied. Please find file ${C.DB_NAME} in external storage", Toast.LENGTH_LONG).show()
    }

    private fun testBenchmark() {
        val startTime = System.nanoTime()
        for (i in 0 until SIZE) {
            logger.d("Logger", "Logger->Message" + i)
        }
        Log.d("LoggerResult", "Logger ->Time:" + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime))

        //        startTime = System.nanoTime();
        //        for (int i = 0; i < SIZE; i++) {
        //            Log.d("Logger", "Default->Message" + i);
        //        }
        //        Log.d("DefaultLoggerResult", "Time:" + TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startTime));
    }

    companion object {
        val SIZE = 1000
    }
}
